BuildClient.hs

```diff
@@ -134,6 +139,23 @@ initialise opts uri auxUris
   where
     readMissingOpt prompt = maybe (putStrLn prompt >> getLine) return
 
+
+-- | Parse the @00-index.cache@ file of the available package repositories.
+parseRepositoryIndices :: IO (S.Set PackageIdentifier)
+parseRepositoryIndices = do
```

Comment:
Note that this will likely break for secure repos as the filename and on-disk format for 01-index.* changed

In future `cabal` versions we should extend or add a `cabal list` variant that dumps the required information (including additional meta-data such as revision number and/or last-revision-timestamp, and maybe also repository provenance) so that we don't need to understand the `.cache` format